src/nvim/move.c

```diff
@@ -2040,14 +2048,14 @@ void halfpage(bool flag, linenr_T Prenum)
 {
   long scrolled = 0;
   int i;
-  int n;
   int room;
 
   if (Prenum)
     curwin->w_p_scr = (Prenum > curwin->w_height) ?
                       curwin->w_height : Prenum;
-  n = (curwin->w_p_scr <= curwin->w_height) ?
-      curwin->w_p_scr : curwin->w_height;
+  assert(curwin->w_p_scr <= INT_MAX);
```

Comment:
`curwin->w_p_scr` corresponds to `scroll` option, which is guaranteed to be strictly positive.
See option.c:5338.